src/Concordium/Client/Runner.hs

```diff
@@ -2064,14 +2069,9 @@ processContractCmd action baseCfgDir verbose backend =
 -- Or, log fatally with appropriate error messages if anything goes wrong.
 getContractInfo :: (MonadIO m) => NamedContractAddress -> Text -> ClientMonad m CI.ContractInfo
 getContractInfo namedContrAddr block = do
-  res <- getInstanceInfo (Text.pack . showCompactPrettyJSON . ncaAddr $ namedContrAddr) block
-  case res of
-    Left err -> logFatal ["I/O error:", err]
-    -- TODO: Handle nonexisting blocks separately from nonexisting contracts.
-    Right (GRPCResponse _ AE.Null) -> logFatal [[i|the contract instance #{showNamedContractAddress namedContrAddr} does not exist in block #{block}|]]
-    Right contrInfo -> case AE.fromJSON $ grpcResponseVal contrInfo of
-      Error err -> logFatal ["Could not decode contract info:", err]
-      Success info -> pure info
+  b <- readOrFail block
+  r <- getResponseValueOrFail =<< getInstanceInfoV2 (ncaAddr namedContrAddr) (Given b)
```

Comment:
What is now printed if the contract doesn't exist?
We should make sure to still print a nice message, as that will happen quite frequently.

Reply:
It prints the node response, in this case it would be:

`Error: A GRPC error occurred: gRPC error: block or instance not found.`

Which is the response from the node. It is not possible to really discern further based on the response.

Reply:
Well it is. And the error as it was before was just better than it is now. So please keep the error as it was, there is no reason to change it.

Further, you have deleted a `TODO` which you could handle here as well. In the error case you could query whether the block exists (e.g., getBlockInfo), and print the corresponding error message.